pyqtgraph/graphicsItems/PlotCurveItem.py

```diff
@@ -15,6 +14,126 @@
 __all__ = ['PlotCurveItem']
 
 
+class GLC:
```

Comment:
This might be a good use-case for a [dataclass](https://docs.python.org/3/library/dataclasses.html) with `frozen=True`